lib/Conversion/TorchToTMTensor/TorchToTMTensor.cpp

```diff
@@ -299,6 +299,54 @@ createTMTensorSortOp(PatternRewriter &rewriter, Location sortOpLoc,
   return SmallVector<Value>(sortOp.getResults());
 }
 
+namespace {
+class ConvertAtenScatterSrcOp : public OpConversionPattern<AtenScatterSrcOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  LogicalResult
+  matchAndRewrite(AtenScatterSrcOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    if (failed(verifyLinalgCompatibleTypes(op, rewriter)))
+      return failure();
+    Location loc = op.getLoc();
+    TypeConverter *typeConverter = getTypeConverter();
+    Value self = adaptor.getSelf();
+    Value index = adaptor.getIndex();
+    Value src = adaptor.getSrc();
+
+    RankedTensorType selfType = self.getType().cast<RankedTensorType>();
+    RankedTensorType indexType = index.getType().cast<RankedTensorType>();
+    RankedTensorType srcType = src.getType().cast<RankedTensorType>();
+    if (selfType.getRank() != indexType.getRank() ||
+        indexType.getRank() != srcType.getRank())
+      return rewriter.notifyMatchFailure(op,
+                                         "'self', 'index' and 'src' should all "
+                                         "have the same number of dimensions.");
+
+    int64_t dim;
+    if (!matchPattern(op.getDim(), m_TorchConstantInt(&dim)))
+      return op.emitError("unimplemented: dim is not constant");
```

Comment:
nit: use `notifyMatchFailure` in `matchAndRewrite` patterns instead of `emitError`